src/components/AssetsComponent.jsx

```diff
@@ -20,46 +21,66 @@ export default class AssetsComponent extends Component {
   };
 
   componentDidMount() {
-    const { activePage, assetsList } = this.props;
-
-    const assetsEmpty = isEmpty(assetsList);
-
     // TODO: fix the logic so that assets are fetched when you create an asset before fetching
     // assets, otherwise, you'll only display 1 row in assets table yet there are more than one
     // assets
-    if (assetsEmpty) {
-      this.props.getAssetsAction(activePage, this.state.limit);
-    }
+    this.loadAssets();
 
     this.props.loadAllAssetModels();
     this.props.loadDropdownAssetTypes();
   }
 
+  loadAssets = () => {
+    const { activePage, assetsList, match } = this.props;
+    const { assetStatus } = match.params;
+
+    const assetsEmpty = isEmpty(assetsList);
+    const assetStat = assetStatus === undefined ? '' : assetStatus;
+    const shouldReload = this.props.assetStatus !== assetStat;
+
+    if (shouldReload) {
+      this.props.getAssetsAction(activePage, this.state.limit, null, assetStatus);
+    }
+
+    if (assetsEmpty) {
+      this.props.getAssetsAction(activePage, this.state.limit, null, assetStatus);
+    }
+  };
+
   handleRowChange = (e, data) => {
+    const { activePage, match } = this.props;
+    const { assetStatus } = match.params;
+
     this.setState({ limit: data.value });
     this.props.resetAssets();
-    this.props.getAssetsAction(this.props.activePage, data.value);
+
+    this.props.getAssetsAction(activePage, data.value, null, assetStatus);

```

Comment:
Hmmm...let's update `loadAssets` so that it can be reused here. This means that after moving `shouldReload` and `assetsEmpty` to the container as explained in a previous comment, we'll have the following:
```jsx
componentDidMount() {
  const { activePage, status } = this.props;
  if (shouldFetchAssets) {
    this.loadAssets(
      activePage,
      this.state.limit,
      null,
      status
    );
  }
  ...
}

loadAssets  = (activePage, limit, filters, status) => {
  return this.props.getAssetsAction(
    activePage,
    limit,
    filters,
    status
  );
}

handleRowChange = (e, data) => {
  ...
  this.loadAssets(activePage, data.value, null, assetStatus);
}